File: 205.py
class Solution:
    def isomorphic(self, s: str, t: str):
        print(self.isIsomorphic(s,t))

    def isIsomorphic(self, s: str, t: str) -> bool:
        sMap = {}
        tMap = {}
        for index in range(len(s)):
            if s[index] not in sMap and t[index] not in tMap:
                sMap[s[index]] = t[index]
                tMap[t[index]] = s[index]
            elif s[index] not in sMap and t[index] in tMap:
                return False
            elif t[index] not in tMap and s[index] in sMap:
                return False
            if sMap[s[index]] != t[index]:
                return False
            if tMap[t[index]] != s[index]:
                return False
        return True

sol = Solution()
sol.isomorphic("egg", "add")
sol.isomorphic("foo", "bar")
sol.isomorphic("paper", "title")
sol.isomorphic("badc", "baba")
sol.isomorphic("egcd", "adfd")

File: 205.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    for(let i=0; i<s.length;i++) {
        s=s.replaceAll(s[i],t[i]);
        if(s === t) return true;
    }
    return false;
};

const test1 = ["egg", "add"];
const test2 = ["foo", "bar"];
const test3 = ["paper", "title"];

console.log(isIsomorphic(test1[0], test1[1]));
console.log(isIsomorphic(test2[0], test2[1]));
console.log(isIsomorphic(test3[0], test3[1]));